lib/ruby_lsp/document.rb

```diff
@@ -89,20 +92,17 @@ def parse
       return if @unparsed_edits.empty?
 
       @unparsed_edits.clear
-      @tree = SyntaxTree.parse(@source)
-      @syntax_error = false
-    rescue SyntaxTree::Parser::ParseError
-      @syntax_error = true
+      @parse_result = YARP.parse(@source)
     end
 
     sig { returns(T::Boolean) }
     def syntax_error?
-      @syntax_error
+      @parse_result.failure?
     end
 
     sig { returns(T::Boolean) }
     def parsed?
-      !@tree.nil?
+      !!@parse_result.value
```

Comment:
Let's get rid of this method, there's no value in keeping it anymore. We parse the document the moment it's instantiated and because YARP always succeeds in returning an AST, this method will always return `true` no matter what.